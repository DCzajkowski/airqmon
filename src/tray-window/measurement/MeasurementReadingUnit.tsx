import * as React from 'react';

export enum Unit {
  PM = 'PM',
  TEMP_C = 'TEMP_C',
  PRESSURE_PA = 'PRESSURE_PA',
  PERCENT = 'PERCENT',
}

export interface IMeasurementReadingUnitProps {
  unit?: Unit;
}

export const MeasurementReadingUnit = ({ unit = null }: IMeasurementReadingUnitProps) => {
  let content = {
    [Unit.PM]: (
      <>
        Î¼g/m<sup>3</sup>
      </>
    ),
    [Unit.TEMP_C]: (
      <>
        <sup>&deg;</sup>C
      </>
    ),
    [Unit.PRESSURE_PA]: 'hPA',
    [Unit.PERCENT]: '%',
  }[unit];

  if (content === undefined) {
    content = null;
  }

  return unit !== null ? <div className="measurement__unit"> {content}</div> : null;
};
