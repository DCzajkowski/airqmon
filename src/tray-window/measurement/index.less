.measurement {
  flex-basis: 50%;
  flex-direction: row;
  margin-top: 8px;

  &__reading,
  &__description {
    text-align: center;
    font-weight: 300;
  }

  &__reading {
    font-size: @font-size + 2pt;
    color: @text-color;
  }

  &__norm {
    font-size: @font-size - 3pt;
    vertical-align: super;

    &::before {
      content: ' ';
      white-space: pre;
    }
  }

  &__unit {
    display: inline;

    &::before {
      content: '';
      white-space: pre;
    }
  }

  &__description {
    text-transform: uppercase;
    color: #666;
    font-size: @font-size - 2pt;
  }
}
