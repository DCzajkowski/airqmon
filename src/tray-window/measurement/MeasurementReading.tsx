import * as React from 'react';

import { MeasurementReadingUnit, IMeasurementReadingUnitProps } from './MeasurementReadingUnit';

export interface IMeasurementReadingProps extends IMeasurementReadingUnitProps {
  reading?: number | string;
  formatter?: (val: number | string) => string;
}

export const MeasurementReading: React.SFC<IMeasurementReadingProps> = ({
  reading,
  formatter,
  unit,
  children,
}) => {
  const content = reading ? (formatter ? formatter(reading) : reading) : 'n/a';

  return (
    <>
      <div className="measurement__reading">
        {content}
        {reading ? <MeasurementReadingUnit unit={unit} /> : null}
        {children}
      </div>
    </>
  );
};
