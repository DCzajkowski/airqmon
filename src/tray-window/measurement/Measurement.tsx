import * as React from 'react';

import { MeasurementReading, IMeasurementReadingProps } from './MeasurementReading';

export const formatters: { [key: string]: (val: string | number) => string } = {
  toFixed2: (val: number) => val.toFixed(2),
  divBy100ToFixed2: (val: number) => (val / 100).toFixed(2),
};

export interface IMeasurementProps extends IMeasurementReadingProps {
  description: string;
}

export const Measurement = ({ description, unit, reading, formatter }: IMeasurementProps) => {
  return (
    <div className="measurement">
      <MeasurementReading reading={reading} unit={unit} formatter={formatter} />
      <div className="description">{description}</div>
    </div>
  );
};
