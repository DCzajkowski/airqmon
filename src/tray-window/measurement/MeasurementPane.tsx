import * as React from 'react';

import { IAirlyCurrentMeasurement } from '../../airly';
import { Contaminates } from '../../contamination';
import { formatters, Measurement } from './Measurement';
import { Unit } from './MeasurementReadingUnit';

interface IMeasurementPaneProps {
  measurement?: IAirlyCurrentMeasurement;
}

const MeasurementPane = ({ measurement }: IMeasurementPaneProps) => {
  return (
    <div className="measurement-pane">
      <Measurement
        contaminate={Contaminates.PM25}
        reading={measurement.pm25}
        formatter={formatters.significant}
      />
      <Measurement
        contaminate={Contaminates.PM10}
        reading={measurement.pm10}
        formatter={formatters.significant}
      />
      <Measurement
        contaminate={Contaminates.PM1}
        reading={measurement.pm1}
        formatter={formatters.significant}
      />
      <Measurement
        description="Temperature"
        unit={Unit.TEMP_C}
        reading={measurement.temperature}
        formatter={formatters.toFixed2}
      />
      <Measurement
        description="Pressure"
        unit={Unit.PRESSURE_PA}
        reading={measurement.pressure / 100}
        formatter={formatters.toFixed1}
      />
      <Measurement
        description="Humidity"
        unit={Unit.PERCENT}
        reading={measurement.humidity}
        formatter={formatters.toFixed2}
      />
    </div>
  );
};

export default MeasurementPane;
