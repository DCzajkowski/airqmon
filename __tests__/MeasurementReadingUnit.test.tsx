import * as React from 'react';
import { shallow } from 'enzyme';
import { Unit, MeasurementReadingUnit } from '../src/tray-window/measurement/MeasurementReadingUnit'

test('MeasurementReadingUnit renders proper unit for Unit.PERCENT', () => {
  const wrapper = shallow(<MeasurementReadingUnit unit={Unit.PERCENT} />);
  expect(wrapper.text()).toMatchSnapshot();
});

test('MeasurementReadingUnit renders proper unit for Unit.PM', () => {
  const wrapper = shallow(<MeasurementReadingUnit unit={Unit.PM} />);
  expect(wrapper.text()).toMatchSnapshot();
});

test('MeasurementReadingUnit renders proper unit for Unit.PRESSURE_PA', () => {
  const wrapper = shallow(<MeasurementReadingUnit unit={Unit.PRESSURE_PA} />);
  expect(wrapper.text()).toMatchSnapshot();
});

test('MeasurementReadingUnit renders proper unit for Unit.TEMP_C', () => {
  const wrapper = shallow(<MeasurementReadingUnit unit={Unit.TEMP_C} />);
  expect(wrapper.text()).toMatchSnapshot();
});

test('MeasurementReadingUnit doesn\'t render if no unit is given', () => {
  const wrapper = shallow(<MeasurementReadingUnit />);
  expect(wrapper.getElement()).toBeNull();
});

