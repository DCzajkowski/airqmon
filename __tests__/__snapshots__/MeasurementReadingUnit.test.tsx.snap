// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeasurementReadingUnit renders proper unit for Unit.PERCENT 1`] = `" %"`;

exports[`MeasurementReadingUnit renders proper unit for Unit.PM 1`] = `" μg/m3"`;

exports[`MeasurementReadingUnit renders proper unit for Unit.PRESSURE_PA 1`] = `" hPA"`;

exports[`MeasurementReadingUnit renders proper unit for Unit.TEMP_C 1`] = `" °C"`;
